src/devices/Card/Ndef/samples/Program.cs

```diff
@@ -0,0 +1,442 @@
+    var debugLevelConsole = Console.ReadKey();
+    Console.WriteLine();
+    LogLevel debugLevel = debugLevelConsole is { KeyChar: 'Y' or 'y' } ? LogLevel.Debug : LogLevel.None;
+
+    if (choiceInterface is { KeyChar: '3' })
+    {
+        Console.WriteLine("Which pin number do you want as Chip Select?");
+        var pinSelectConsole = Console.ReadLine();
+        int pinSelect;
+        try
+        {
+            pinSelect = Convert.ToInt32(pinSelectConsole);
+        }
+        catch (Exception ex) when (ex is FormatException || ex is OverflowException)
+        {
+            Console.WriteLine("Impossible to convert the pin number.");
+            return;
+        }
+
+        pn532 = new Pn532(SpiDevice.Create(new SpiConnectionSettings(0)), pinSelect, logLevel: debugLevel);
+    }
+    else if (choiceInterface is { KeyChar: '2' })
+    {
+        pn532 = new Pn532(I2cDevice.Create(new I2cConnectionSettings(1, Pn532.I2cDefaultAddress)), debugLevel);
+    }
+    else
+    {
+        Console.WriteLine("Please enter the serial port to use. ex: COM3 on Windows or /dev/ttyS0 on Linux");
+
+        var device = Console.ReadLine();
+        pn532 = new Pn532(device!, debugLevel);
+    }
+
+    nfcReader = pn532;
+    if (pn532.FirmwareVersion is FirmwareVersion version)
+    {
+        Console.WriteLine($"Is it a PN532!: {version.IsPn532}, Version: {version.Version}, Version supported: {version.VersionSupported}");
+    }
+    else
+    {
+        Console.WriteLine($"Error connecting to PN532");
+        return;
+    }
+
+    // Let's pull for 20 seconds and see the result
+    Console.WriteLine("Place a type A card on reader");
+    byte[]? retData = null;
+    while ((!Console.KeyAvailable))
+    {
+        retData = pn532.ListPassiveTarget(MaxTarget.One, TargetBaudRate.B106kbpsTypeA);
+        if (retData is object)
+        {
+            break;
+        }
+
+        // Give time to PN532 to process
+        Thread.Sleep(200);
+    }
+
+    if (retData is null)
+    {
+        return;
+    }
+
+    Console.WriteLine();
+
+    var decrypted = pn532.TryDecode106kbpsTypeA(retData.AsSpan().Slice(1));
+    if (decrypted is not object)
+    {
+        Console.WriteLine("Can't read properly the card");
+        return;
+    }
+
+    RunTestNdef(pn532, decrypted);
+}
+
+void RunTestNdef(CardTransceiver transceiver, Data106kbpsTypeA card)
+{
+    Console.WriteLine($"ATQA: {card.Atqa}");
+    Console.WriteLine($"SAK: {card.Sak}");
+    Console.WriteLine($"UID: {BitConverter.ToString(card.NfcId)}");
+
+    MifareCard mifareCard = new MifareCard(transceiver, card.TargetNumber) { BlockNumber = 0, Command = MifareCardCommand.AuthenticationA };
+    mifareCard.SetCapacity(card.Atqa, card.Sak);
+    mifareCard.SerialNumber = card.NfcId;
+    mifareCard.KeyA = new byte[6] { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+    mifareCard.KeyB = new byte[6] { 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF };
+
+    if (testToRun.KeyChar == '1')
```

Comment:
this could use switch statement

Reply:
ho yes, will do, it's more clear. Will do it where it makes sense as well